code/game/objects/items/weapons/tools.dm

```diff
@@ -241,12 +241,20 @@
 		if(!(S.status & ORGAN_ROBOT) || user.a_intent != I_HELP || S.open == 2)

```

Comment:
THIS

is the line that's causing people to get hurt when self-repairing.

Granted, this would also provide an imperative for robotic repair procedures to actually be done correctly.